samples/wifi/sta/src/main.c

```diff
@@ -317,3 +319,25 @@ int main(void)
 
 	return 0;
 }
+
+void wifi_ready_cb(void)
+{
+	LOG_INF("Wi-Fi is ready");
+	k_sem_give(&wifi_ready_sem);
```

Comment:
Handling Wi-Fi ready at boot is trivial, the complex part is handling Wi-Fi not ready as it's runtime, so, we should add that.

Reply:
Handled Wi-Fi reconnection for the time being. Will check and enhance later.